communication/communications.py

```diff
@@ -77,8 +78,30 @@ def _write(self, sig, opcode, arg, seqNo):
         Sends the code to the Arduino
         """
         checksum = self.create_checksum(arg, opcode)
-        opcode_string = "%d%d%03d%d\r" % (sig, opcode, arg, checksum)
+        opcode_string = "%d%d%03d%d%d\r" % (sig, opcode, arg, checksum, seqNo)
         self.to_robot(opcode_string)
+        
+        # Check twice if response received, if not - resend
+        success = False
+        while (!success):
+            for i in range(2):
+                sleep(0.1)
+                response = self.internal_queue.peek()
+                if response:
+                    # Check if success
+                    # (not checksum fail or unrecognized or bad command length)
+                    # Note: would be better to have a specific response upon success
+                    if response not in ["0CF","0UC", "OIW"] and
```

Comment:
The idea is,  if I reach success,  I stop attempting to re-send. And I may be wrong,  but I suspected the read/write streams pop the queue anyway after doing a read/write.  I didn't remove any pops added by you,  so I thought that pops happen at a higher level (will reread code to confirm)